cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseQueue.java

```diff
@@ -189,4 +195,30 @@ public boolean isShutdown() {
             throw new RejectedExecutionException("Database is closed");
         }
     }
+
+    /**
+     * Returns the SQLite Version.
+     * @return The SQLite version or "Unknown" if the version could not be determined.
+     */
+    public String getSQLiteVersion() {
```

Comment:
Save the result to a field for now - although I suspect retrieving the version to be much faster than a query because it is probably just a constant we should avoid the need to go native every time we ask for it.

We could consider moving extraction of the version into the validation as part of #292 and having it available from the `SQLDatabase` as well, which would open up opportunities for us to add the version to exceptions at a lower level, something to consider while we figure out the places it would be useful to record it.